Actions/settingsActions.js

```diff
@@ -44,14 +49,21 @@ export const loadSettings = deviceId => {
 
 export const storeSettings = (deviceId, newSettings) => {
   return dispatch => {
+    dispatch(storeSettingsBegin());
     fetch(`pathToOurSettingsApi/${deviceId}`, {
       method: "PUT",
       headers: {
         "Content-Type": "application/json"
       },
       body: JSON.stringify(newSettings)
     })
-      .then(res => res.json())
+      .then(res => {
+        if(!res.ok)
+        {
+          throw Error(res.statusText);
```

Comment:
don't throw error
create action storeSettingsFail and set information about error in state